src/api/remote.rs

```diff
@@ -254,6 +364,13 @@ impl Reference {
             Reference::Git(git) => git.name().as_str(),
         }
     }
+
+    /// The reference's type (branch/tag)
+    pub fn reference_type(&self) -> &git::Reference {
+        match self {
+            Reference::Git(reference) => reference,
+        }
+    }
```

Comment:
This function is generically named, but only applies to Git references. Prefer to make the caller access this directly.

In particular other remotes may not have the concept of "branches" and "tags".